src/hooks.ts

```diff
@@ -293,18 +326,25 @@ export const useExperiment: UseExperiment = (experimentKey, options = {}, overri
       });
     }
     return (): void => {};
-  }, [optimizely.getIsReadyPromiseFulfilled(), options.autoUpdate, optimizely, experimentKey, getCurrentDecision]);
+  }, [isReadyPromiseFulfilled, options.autoUpdate, optimizely, experimentKey, getCurrentDecision]);
 
   useEffect(
     () =>
-      optimizely.onForcedVariationsUpdate(() => {
+      optimizely?.onForcedVariationsUpdate(() => {
         setState((prevState) => ({
           ...prevState,
           ...getCurrentDecision(),
         }));
       }),
     [getCurrentDecision, optimizely]
   );
+
+  if (!optimizely) {
+    hooksLogger.error(
+      `Unable to use experiment ${experimentKey}. optimizely prop must be supplied via a parent <OptimizelyProvider>`
```

Comment:
NIT: Can we move ``` The `optimizely` prop must be supplied via a parent <OptimizelyProvider>.``` as a `const` in this file?

Reply:
Sure! Great suggestion!